src/xdp-utils.c

```diff
@@ -144,11 +149,79 @@ xdp_app_info_new (XdpAppInfoKind kind)
   return app_info;
 }
 
+char *
+_xdp_parse_app_id_from_unit_name (const char *unit)
+{
+  g_autoptr(GRegex) regex = NULL;
+  g_autoptr(GMatchInfo) match = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *app_id = NULL;
+
+  g_assert (g_str_has_prefix (unit, "app-"));
+
+  /*
+   * From https://systemd.io/DESKTOP_ENVIRONMENTS/ the format is one of:
+   * app[-<launcher>]-<ApplicationID>-<RANDOM>.scope
+   * app[-<launcher>]-<ApplicationID>-autostart.service -> no longer true since systemd v248
+   * app[-<launcher>]-<ApplicationID>[@<RANDOM>].service
```

Comment:
The 2 formats above aren't tested in the test suite.

Reply:
I'm not sure what you mean?
- `app[-<launcher>]-<ApplicationID>-<RANDOM>.scope` tested by `app-gnome-org.gnome.Epiphany-182352.scope`
- `app[-<launcher>]-<ApplicationID>-autostart.service` tested by `app-gnome-org.gnome.SettingsDaemon.DiskUtilityNotify-autostart.service`
- `app[-<launcher>]-<ApplicationID>[@<RANDOM>].service` tested by `app-KDE-org.kde.okular@12345.service`

Reply:
I'm not sure how I came to that conclusion when I was reading the diff, but I was obviously not concentrating hard enough.